packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -230,6 +240,10 @@ export async function deployStack(options: DeployStackOptions): Promise<DeploySt
     ? templateParams.updateExisting(finalParameterValues, cloudFormationStack.parameters)
     : templateParams.supplyAll(finalParameterValues);
 
+  if (options.hotswapLogMonitor) {
+    await registerCloudWatchLogGroups(options.sdkProvider, stackArtifact, assetParams, options.hotswapLogMonitor);
+  }
```

Comment:
So, this means we will re-do this scanning every time a deployment happens.

Doesn't that kind of defeat the purpose of re-using this `hotswapLogMonitor`? We might as well return a new one from `registerCloudWatchLogGroups()`, right?

Also, I don't think we ever remove Log Groups from the monitor, right? Which means that, if I make a deployment that removes a Log Group, or a Lambda, we will still poll it for changes (and I assume print out a warning about not being able to find it).

This is probably another good reason to return a new monitor from `registerCloudWatchLogGroups()`, instead of re-using the same one.

Reply:
I changed the `addLogGroups` method to be `setLogGroups` to hopefully address this. I think we do need to re-scan after every deployment for the cases you mentioned (adding/removing log groups) so now we just do the evaluation after every deployment and replace our list of log groups that we are monitoring. I had originally used `addLogGroups` because I was thinking of being able to add groups at different places, but now we are just doing the evaluation 1 time for each deployment.

Reply:
Actually, I think this doesn't work for multiple stacks.

Reply:
OK I now have both a `reset` and `add` methods on the log monitor. Before each deployment we `reset` the log groups and then each stack registers it's log groups by calling the `add` method. This way we can keep track of added/removed log groups with multiple stacks.